apps/wpcom-block-editor/src/calypso/features/iframe-bridge-server.js

```diff
@@ -676,15 +676,27 @@ function openLinksInParentFrame( calypsoPort ) {
 		} );
 	} );
 
-	if ( calypsoifyGutenberg.manageReusableBlocksUrl ) {
-		const manageReusableBlocksLinkSelectors = [
-			'.block-editor-inserter__manage-reusable-blocks', // Link in the Blocks Inserter
-			'a.components-menu-item__button[href*="post_type=wp_block"]', // Link in the More Menu
-		].join( ',' );
-		$( '#editor' ).on( 'click', manageReusableBlocksLinkSelectors, ( e ) => {
-			e.preventDefault();
-			window.open( calypsoifyGutenberg.manageReusableBlocksUrl, '_top' );
-		} );
+	// Create a new post link in block settings sidebar for Query block
+	if ( calypsoifyGutenberg.createNewPostUrl || calypsoifyGutenberg.manageReusableBlocksUrl ) {
+		const $editor = $( '#editor, #edit-site-editor' );
+		const replaceLink = () => {
+			if ( calypsoifyGutenberg.createNewPostUrl ) {
+				$( '.wp-block-query__create-new-link a', $editor )
+					.attr( 'href', calypsoifyGutenberg.createNewPostUrl )
+					.attr( 'target', '_top' );
+			}
+
+			if ( calypsoifyGutenberg.manageReusableBlocksUrl ) {
+				const manageReusableBlocksLinkSelectors = [
+					'.block-editor-inserter__manage-reusable-blocks', // Link in the Blocks Inserter
+					'a.components-menu-item__button[href*="post_type=wp_block"]', // Link in the More Menu
+				].join( ',' );
+				$( manageReusableBlocksLinkSelectors, $editor )
+					.attr( 'href', calypsoifyGutenberg.manageReusableBlocksUrl )
+					.attr( 'target', '_top' );
+			}
+		};
+		setInterval( replaceLink, 50 );
```

Comment:
I'm curious about this setInterval. Are there any events we could listen to instead?

Reply:
The trouble with the previous approach is that intercepting the click event won't handle "open in a new tab" flows, where the browser takes the href and opens the url in a new tab. Without changing the actual href, there's no effect on the destination url.

One solution might be using a mutation observer. That feels heavy, but it's likely worth it. I think opening a new tab is going to be common with this particular link, because otherwise you need to save your changes before you exit the editor to create a new post. cc @david-szabo97

Reply:
Ah I see. One note about the mutation observer -- if we end up going down that route, let's be sure to scope to a specific enough target when calling `mutationObserver.observe()`. 

A commit to the coblocks library actually borked the block editor because the changes added a condition whereby, instead of just a subset of the DOM document, the entire DOM document was tracked for changes. ( see pbAok1-1e6-p2 under subheading "Root Cause" )

Reply:
That was tough to pull off, but it's ready for review 😄